File: app/src/main/java/com/attendo/data/SubDao.java
package com.attendo.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface SubDao {

    @Insert
    void insert(SubEntity SUBJECT);

    @Query("SELECT * FROM SubjectName")
    LiveData<List<SubEntity>> getAllSubjects();

    @Delete
    int delete(SubEntity SUBJECT);

    @Query("UPDATE SubjectName SET present = :present WHERE id = :id")
    void updatePresent(int present, int id);

    @Query("UPDATE SubjectName SET absent = :absent WHERE id = :id")
    void updateAbsent(int absent, int id);


    @Query("UPDATE SubjectName SET total= :total WHERE id = :id")
    void updateTotal(int total, int id);

    @Query("UPDATE SubjectName SET subject = :subject WHERE id = :id")
    void updateSubject(String subject,int id);



}

File: app/src/main/java/com/attendo/data/sub/SubDao.kt
package com.attendo.data.sub

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao

interface SubDao {

    @Insert
    fun insert(SUBJECT: SubEntity?)

    @Query("SELECT * FROM SubjectName")
    fun getAllSubjects(): LiveData<List<SubEntity?>?>?

    @Delete
    fun delete(SUBJECT: SubEntity?): Int

    @Query("UPDATE SubjectName SET present = :present WHERE id = :id")
    fun updatePresent(present: Int, id: Int)

    @Query("UPDATE SubjectName SET absent = :absent WHERE id = :id")
    fun updateAbsent(absent: Int, id: Int)


    @Query("UPDATE SubjectName SET total= :total WHERE id = :id")
    fun updateTotal(total: Int, id: Int)

    @Query("UPDATE SubjectName SET subject = :subject WHERE id = :id")
    fun updateSubject(subject: String?, id: Int)


}